docs/edgeql/group.rst

```diff
@@ -0,0 +1,195 @@
+.. _ref_eql_group:
+
+Group
+=====
+
+EdgeQL supports a top-level ``group`` statement. These are used to partition
+sets into subsets based on some parameters. These subsets then can be
+additionally aggregated to provide some analytics.
+
+The most basic format is just using the bare :eql:stmt:`group` to group a set
+of objects by some property:
+
+.. code-block:: edgeql-repl
+
+    db> with M := (select Movie filter .release_year > 2015)
```

Comment:
I've moved it further down in examples and prefaced it with actual motivation (you want to only group a subset of all data). Hopefully this works. Limiting input set is important for grouping by multiple parameters because otherwise it kinda blows up with all the combinations.